homeassistant/components/weather/__init__.py

```diff
@@ -986,3 +1015,50 @@ def async_registry_entry_updated(self) -> None:
                 )
             ) and custom_unit_visibility in VALID_UNITS[ATTR_WEATHER_VISIBILITY_UNIT]:
                 self._weather_option_visibility_unit = custom_unit_visibility
+
+    @final
+    @callback
+    def async_subscribe_forecast(
+        self,
+        forecast_type: Literal["daily", "hourly", "twice_daily"],
+        forecast_listener: Callable[[list[dict[str, Any]] | None], None],
+    ) -> CALLBACK_TYPE:
+        """Subscribe to forecast updates."""
+        self._forecast_listeners[forecast_type].append(forecast_listener)
+
+        @callback
+        def unsubscribe() -> None:
+            self._forecast_listeners[forecast_type].remove(forecast_listener)
+
+        return unsubscribe
+
+    @final
+    async def async_update_forecast(
```

Comment:
This name is a bit unclear /  misleading, should it be renamed to `async_forecast_updated` or `async_push_forecast`?

Reply:
I do agree the current method name is misleading.

Maybe just `update_listeners`? wich is more in line with what we use elsewhere? Otherwise `async_forecast_updated` sounds ok to.